illumos-utils/src/zfs.rs

```diff
@@ -203,14 +207,84 @@ pub struct EncryptionDetails {
 
 #[derive(Debug, Default)]
 pub struct SizeDetails {
-    pub quota: Option<usize>,
-    pub reservation: Option<usize>,
+    pub quota: Option<ByteCount>,
+    pub reservation: Option<ByteCount>,
     pub compression: CompressionAlgorithm,
 }
 
+#[derive(Debug)]
+pub struct DatasetProperties {
+    /// The Uuid of the dataset
+    pub id: Option<DatasetUuid>,
+    /// The full name of the dataset.
+    pub name: String,
+    /// Remaining space in the dataset and descendents.
+    pub avail: ByteCount,
+    /// Space used by dataset and descendents.
+    pub used: ByteCount,
+    /// Maximum space usable by dataset and descendents.
+    pub quota: Option<ByteCount>,
+    /// Minimum space guaranteed to dataset and descendents.
+    pub reservation: Option<ByteCount>,
+    /// The compression algorithm used for this dataset.
+    pub compression: String,
```

Comment:
Does `compression` need to be a more specific type than this? If not then could you add a comment? (I guess `compression` is not something we parse?)

Reply:
There is a type in `omicron_common::disk::CompressionAlgorithm` that is the strongly-typed version of this -- I'll link to it in a comment here -- but I didn't want to rely on one of those algorithms being known (nor did I really want to add an `Other(...)` variant) for the inventory to be collected.

Here's my fear: a new Helios update adds a new compression algorithm that the sled agent doesn't know about, and tries to set that value on the root filesystem. If this happens, I don't want a parsing error to prevent inventory collection.

Reply:
Yep, reasonable